alica_engine/include/engine/model/Plan.h

```diff
@@ -19,13 +19,16 @@ class PreCondition;
 class RuntimeCondition;
 class AlicaEngine;
 class BlackboardBlueprint;
+class ConfigChangeListener;
 /**
  * An ALICA plan
  */
 class Plan : public AbstractPlan
 {
 public:
-    Plan(AlicaEngine* ae, int64_t id);
+    //[[deprecated("It will be removed in the last PR")]]
+    Plan(AlicaEngine* ae, int64_t id); // TOBE removed
+    Plan(const YAML::Node& config, ConfigChangeListener& configChangeListener, int64_t id);
```

Comment:
to further simplify, config could be part of ConfigChangeListener and could be exposed via a getConfig() method.

Reply:
Done